Enable Material 3 on `background_isolate_channels`

miquelbeltran opened the pull request:
Enabling Material 3 on `background_isolate_channels` sample

#### Before

<img width="912" alt="Screenshot 2023-07-20 at 13 41 54" src="https://github.com/flutter/samples/assets/2494376/f07aee8d-93da-4d5b-af26-acc72ecd48e7">

#### With Material 3

<img width="912" alt="Screenshot 2023-07-20 at 13 42 12" src="https://github.com/flutter/samples/assets/2494376/d160db4c-a303-4d99-acd5-916d586a8519">

## Pre-launch Checklist

- [ ] I read the [Flutter Style Guide] _recently_, and have followed its advice.
- [ ] I signed the [CLA].
- [ ] I read the [Contributors Guide].
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-devrel channel on [Discord].

<!-- Links -->
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[CLA]: https://cla.developers.google.com/
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[Contributors Guide]: https://github.com/flutter/samples/blob/main/CONTRIBUTING.md

domesticmouse commented:
In Material 3 the search area is getting lost. Can we migrate to using the [Material 3 SearchBar](https://api.flutter.dev/flutter/material/SearchBar-class.html)? Here's a [nice intro tutorial](https://maneesha-erandi.medium.com/flutter-ui-essentials-material-3-searchbar-6c1acc540a2) by Maneesha Erandi.

domesticmouse reviewed (changes requested):
Needs a M3 SearchBar

miquelbeltran commented:
<img width="912" alt="Screenshot 2023-07-24 at 13 57 41" src="https://github.com/flutter/samples/assets/2494376/0ed740d9-c47b-4437-a00b-be2de8d308f6">
<img width="912" alt="Screenshot 2023-07-24 at 13 57 34" src="https://github.com/flutter/samples/assets/2494376/3d6a39b8-161b-4388-b16b-d1bd378e25eb">

Replaced the Text edit with a `SearchBar`, moved to the `AppBar` bottom section, and simplified the body of the `Scaffold`
